src/routes.js

```diff
@@ -122,6 +124,13 @@ module.exports = [
     }
   },
   {
+    method: 'POST',
+    path: '/finished',
+    handler: (request, reply) => {
+      reply(request.payload.test ? '' : request.payload);
```

Comment:
hmm.. As far as I can see the only place test is set is in your test. Not sure what you gain by not sending a response.

Also, in node, you have access to `process.env.NODE_ENV` which gets set to `production` somewhere like heroku.

Although this kind of thing should not be utilised at handler level, where I think it is best to keep business logic only.